core-shaded/pom.xml

```diff
@@ -28,97 +28,197 @@
-                    <!-- netty's shading of jctools does not remove its pom files -->
-                    <resource>META-INF/maven/org.jctools/jctools-core/pom.properties</resource>
-                    <resource>META-INF/maven/org.jctools/jctools-core/pom.xml</resource>
-                  </resources>
-                </transformer>
-                <!-- Pick up the alternate manifest that was extracted from the core module -->
-                <transformer implementation="org.apache.maven.plugins.shade.resource.IncludeResourceTransformer">
-                  <resource>META-INF/MANIFEST.MF</resource>
-                  <file>${project.build.directory}/META-INF-shaded/MANIFEST.MF</file>
-                </transformer>
-              </transformers>
-              <!-- Keep the dependencies of driver-core -->
-              <promoteTransitiveDependencies>true</promoteTransitiveDependencies>
+              <instructions>
+                <Bundle-SymbolicName>com.datastax.oss.driver.core</Bundle-SymbolicName>
+                <DynamicImport-Package>*</DynamicImport-Package>
+                <!--
+                Don't include because they aren't OSGi bundles, and the driver bundle can
+                live without them:
+                 - jnr
+                 - net.jcip.annotations
+                 - edu.umd.cs.findbugs.annotations
+                Don't include because they are imported by shaded Netty classes
+                but not used by the driver bundle:
+                 - com.google.protobuf
+                 - com.jcraft.jzlib
+                 - com.ning.compress
+                 - lzma.sdk
+                 - net.jpountz.xxhash
+                 - org.bouncycastle
+                 - org.conscrypt
+                 - org.apache.commons.logging
+                 - org.apache.log4j
+                 - org.apache.logging.log4j
+                 - org.eclipse.jetty
+                 - org.jboss.marshalling
+                 - sun.misc
+                 - sun.security
+                -->
+                <Import-Package>
+                  !com.datastax.oss.driver.shaded.*,
+                  !jnr.*,
+                  !net.jcip.annotations.*,
+                  !edu.umd.cs.findbugs.annotations.*,
+                  !com.google.protobuf.*,
+                  !com.jcraft.jzlib.*,
+                  !com.ning.compress.*,
+                  !lzma.sdk.*,
+                  !net.jpountz.xxhash.*,
+                  !org.bouncycastle.*,
+                  !org.conscrypt.*,
+                  !org.apache.commons.logging.*,
+                  !org.apache.log4j.*,
+                  !org.apache.logging.log4j.*,
+                  !org.eclipse.jetty.*,
+                  !org.jboss.marshalling.*,
+                  !sun.misc.*,
+                  !sun.security.*,
+                  *
+                </Import-Package>
+                <Export-Package>
+                  com.datastax.oss.driver.*.core.*,
```

Comment:
Note that we need to export not only API classes, but also internal driver classes and shaded classes (one reason being that the DSE driver needs them). I think it's a bit unfortunate to expose so many packages, but we don't have the choice.